tests/libnet/cloudinit.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package libnet
+
+import (
+	"fmt"
+	"strings"
+)
+
+type CloudInitNetworkData struct {
+	Ethernets []CloudInitInterface
+}
+
+type CloudInitInterface struct {
+	Addresses   []string
+	Dhcp4       string
+	Gateway6    string
+	Nameservers CloudInitNameservers
+}
+
+type CloudInitNameservers struct {
+	Addresses []string
+	Search    []string
+}
+
+const (
+	ipv6MasqueradeAddress = "fd10:0:2::2/120"
+	ipv6MasqueradeGateway = "fd10:0:2::1"
+)
+
+const cloudInitNetworkHead = `
+version: 2
+ethernets:`
+
+// CreateDefaultCloudInitNetworkData generates a default configuration
+// for the Cloud-Init Network Data, in version 2 format.
+// The default configuration sets dynamic IPv4 (DHCP) and static IPv6 addresses,
+// inclusing DNS settings of the cluster nameserver IP and search domains.
+func CreateDefaultCloudInitNetworkData() (string, error) {
+	dnsServerIP, err := ClusterDNSServiceIP()
+	if err != nil {
+		return "", err
+	}
+
+	networkData := CreateCloudInitNetworkData(
+		CloudInitNetworkData{
+			Ethernets: []CloudInitInterface{
+				{
+					Addresses: []string{ipv6MasqueradeAddress},
+					Dhcp4:     "true",
+					Gateway6:  ipv6MasqueradeGateway,
+					Nameservers: CloudInitNameservers{
+						Addresses: []string{dnsServerIP},
+						Search:    SearchDomains(),
+					},
+				},
+			},
+		},
+	)
+	return networkData, nil
+}
+
+// CreateCloudInitNetworkData generates a configuration for the Cloud-Init Network Data version 2 format
+// based on the inputed data.
+// Supported congifiguration includes IPv4/6 static addresses, IPv4 DHCP and nameservers/dns.
+func CreateCloudInitNetworkData(networkData CloudInitNetworkData) string {
+	data := cloudInitNetworkHead
+
+	ifaceFormat := `
+    eth%d:
+        addresses: [ %s ]
+        dhcp4: %s
+        gateway6: %s
+        nameservers:
+            addresses: [ %s ]
+            search: [ %s ]`
```

Comment:
Are we going to also remove the infamous  configureIPv6OnVMI from https://github.com/kubevirt/kubevirt/blob/master/tests/login.go#L151 and friends ?

Reply:
We should do that for all Fedora images. The Cirros and Alpine do not have a network data option AFAIK.
Maybe we can create a user data instead to do the same on those images.